File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main{
    public static int N, M, V;


    public static boolean[] visited;
    public static LinkedList<Integer>[] adj;
    public static Queue<Integer> queue;

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");

        //문제 조건에 맞게 Graph 그리기
        N = Integer.parseInt(st.nextToken());
        M = Integer.parseInt(st.nextToken());
        V = Integer.parseInt(st.nextToken());

        adj = new LinkedList[N + 1];
        visited = new boolean[N + 1];
        for (int i = 1; i <= N; i++) {
            adj[i] = new LinkedList<Integer>();
        }

        for (int i = 1; i <= M; i++) {

            st = new StringTokenizer(br.readLine(), " ");
            int start = Integer.parseInt(st.nextToken());
            int end = Integer.parseInt(st.nextToken());
            adj[start].add(end);
            adj[end].add(start);
        }
        //ADJ 오름차순으로 정렬해야함

        for (int i = 1; i <= N; i++) {
           Collections.sort(adj[i]);
        }

        //BFS ,DFS 작성해야함
        DFS(V);
        visited = new boolean[N+1];
        System.out.println();
        BFS(V);
    }

    /*
     * DFS 재귀적 방식으로 구현
     * Params : 시작 정점 v
     * */
    public static void DFS(int v) {
        // 방문한

        if(visited[v]){
            return ;
        }
        visited[v]=true;
        System.out.print(v+" ");
        for(int vertex :adj[v]){
            DFS(vertex);
        }
    }

    public static void BFS(int v){
        queue=new LinkedList<>();

        queue.offer(v);
        while(!queue.isEmpty()){
            int vertex =queue.poll();

            // 방문한 정점이라면 다음 정점이동
            if(visited[vertex]){
                continue;
            }
            // 방문한 정점이 아니라면, 방문처리를 함.
            visited[vertex]=true;
            System.out.print(vertex+" ");
            for(int newV : adj[vertex]){
                queue.offer(newV);
            }

        }
    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
from collections import deque

input = sys.stdin.readline


def dfs(visited, node):
    global graph

    if visited[node]:
        return
    visited[node] = True
    print(node, end=" ")

    for next_node in graph[node]:
        dfs(visited, next_node)

def bfs(visited, node):

    queue = deque()
    queue.append(node)

    while queue:
        next_node = queue.popleft()
        if visited[next_node] is True:
            continue
        visited[next_node] = True
        print(next_node, end=" ")
        for linked_node in graph[next_node]:
            queue.append(linked_node)

def solution():
    global graph
    node_num, line_num, start_node = map(int, input().split())

    graph = [[] for _ in range(node_num + 1)]
    for i in range(line_num):
        start, end = map(int, input().split())
        graph[start].append(end)
        graph[end].append(start)
    for i in graph:
        i.sort()
    
    dfs([False] * (node_num + 1), start_node)
    print()
    bfs([False] * (node_num + 1), start_node)

    
if __name__ == "__main__":
    solution()